src/jxl.imageio/jxlinput.cpp

```diff
@@ -0,0 +1,370 @@
+// Copyright Contributors to the OpenImageIO project.
+// SPDX-License-Identifier: Apache-2.0
+// https://github.com/AcademySoftwareFoundation/OpenImageIO
+
+// JPEG XL
+
+// https://jpeg.org/jpegxl/index.html
+// https://jpegxl.info
+// https://jpegxl.info/test-page
+// https://people.csail.mit.edu/ericchan/hdr/hdr-jxl.php
+// https://saklistudio.com/jxltests
+// https://thorium.rocks
+// https://bugs.chromium.org/p/chromium/issues/detail?id=1451807
+
+#include <algorithm>
+#include <cassert>
+#include <cstdio>
+
+#include <OpenImageIO/filesystem.h>
+#include <OpenImageIO/fmath.h>
+#include <OpenImageIO/imageio.h>
+#include <OpenImageIO/tiffutils.h>
+
+#include <jxl/decode.h>
+#include <jxl/decode_cxx.h>
+#include <jxl/resizable_parallel_runner_cxx.h>
+
+OIIO_PLUGIN_NAMESPACE_BEGIN
+
+#define DBG if (0)
+
+class JxlInput final : public ImageInput {
+public:
+    JxlInput() { init(); }
+    ~JxlInput() override { close(); }
+    const char* format_name(void) const override { return "jxl"; }
```

Comment:
Should this be jpeg-xl or jpegxl? We tend to make the "format name" be the name of the format, not necessarily the short file extension. For example, we use "jpeg", "openexr", "tiff", "targa".

Reply:
It should be JPEG XL as written here with non breaking space. No dash.

Reply:
I think I would prefer that it be lower case and not to use a space, since there are circumstances in which the format name is incorporated into filenames.

For that matter, I think all the others make the format_name match the "plugin name", we'll want to make sure this matches the references in imageioplugin.cpp.

I think "jpegxl" is the one you should use. You can't go wrong with being "C identifier safe."

Reply:
We should probably `git mv` the whole directory to be called src/jpegxl.imageio. Then plugin name, short format name, and format_name() will all match, as they do for the rest of the formats.